### PR TITLE
Add x/stakedym

### PR DESCRIPTION
**Pull request by shellvish**
Adds x/stakedym module, tested on local->mainnet flow (staking, unstaking, compounding).

Please note, this does not test granular authz permissions, those should be tested separately.

Also note, we will need to update addresses in `x/stakedym/types/dymension.go`.

Combed through each file, and made sure any reference to tia could be changed without issue. Also high-level traced through logic and think everything should function smoothly.

### CODE REVIEW

File: x/stakedym/types/dymension.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// #nosec G101
+package types
```
Comment (sampocs) [inline x/stakedym/types/dymension.go:2]: verified everything but the addresses. Do you have a doc I can use for the addresses?

File: x/stakedym/types/msgs_test.go [hunk 2]
```diff
@@ -1,950 +0,0 @@
-package types_test
```
Comment (sampocs) [inline x/stakedym/types/msgs_test.go]: did you mean to delete this file?

**Review by sampocs — COMMENTED** [review]
Looks great, thanks for doing this!

Also can you add the store key?

File: x/stakedym/types/dymension.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const (
+	DymensionChainId                   = "dymension_1100-1"
+	StrideToDymensionTransferChannelId = "channel-197"
+	DymensionNativeTokenDenom          = "adym"
+	DymensionNativeTokenIBCDenom       = "ibc/FF6C2E86490C1C4FBBD24F55032831D2415B9D7882F85C3CC9C2401D79362BEA" // #nosec G101
```
Comment (riley-stride) [inline x/stakedym/types/dymension.go:8]:
```
base ❯ dymd q ibc-transfer denom-trace FF6C2E86490C1C4FBBD24F55032831D2415B9D7882F85C3CC9C2401D79362BEA
denom_trace:
  base_denom: ustrd
  path: transfer/channel-0
```

File: x/stakedym/types/msgs_test.go (hunk 2 above)
Comment (shellvish) [reply x/stakedym/types/msgs_test.go]: No, good catch, adding back in!

**Review by asalzmann — APPROVED** [review]

**Pull request merged**